R/PipeOpPredClassifSurv.R

```diff
@@ -0,0 +1,79 @@
+
+PipeOpPredClassifSurv <- R6Class(
+  "PipeOpPredClassifSurv",
+  inherit = PipeOpPredTransformer,
+
+  public = list(
+    initialize = function(id = "trafopred_classifsurv") {
+      super$initialize(id = id,
+                       input = data.table::data.table(name = c("input", "meta"), train = c("NULL", "*"), predict = c("PredictionClassif", "*")),
+                       output = data.table::data.table(name = "output", train = "NULL", predict = "PredictionSurv"))
+    }
+  ),
+
+  private = list(
+    .train = function(input) {
+      self$state = list()
+      list(input)
+    },
+
+    .predict = function(input) {
+      data = input[[2]]
+      pred = input[[1]]
+      data <- cbind(data, pred = pred$prob[, 2])
+
+      ## convert hazards to surv as prod(1 - h(t))
+      surv <- t(sapply(unique(data$id), function(i) {
+        x <- cumprod((1 - data[data$id == i, "pred"]))
+        x
+      }))
+
+      r <- list()
+      time <- sort(unique(data$tend))
+      ## coerce to distribution and crank
+      r <- .surv_return(time, surv = surv)
+
+      crank = c()
+      distr = list()
+
+      # select real tend values by only selecting the last row of each id
+      # basically a slightly more complex unique()
+      real_tend <- data$time2[seq_len(nrow(data)) %% sum(data$id == 1) == 0]
+
+      ## create prediction object
+      data = data %>% group_by(id) %>% filter(tend == max(tend))
+      p <- PredictionSurv$new(
+        row_ids = seq(nrow(data)),
+        crank = r$crank, distr = r$distr,
+        truth = Surv(real_tend, data[["ped_status"]]))
+      ## evaluate with Harrell's C and IGS
+      list(p)
+    }
+  )
+)
+
+task = tsk("rats")
```

Comment:
all the rest of the code you put it in a `test_pipeopxyz.R` in the root of `mlr3proba` for people to test the pipeop or even better in a test file in `tests` (since you will do this anyway)